proxy_connect.go

```diff
@@ -93,6 +98,11 @@ func (proxy *proxy) nextCONNECT(downstream net.Conn) filters.Next {
 		// for example, if some proxy servers reside in jurisdictions where an
 		// origin site is blocked but other proxy servers don't.
 		resp, nextCtx = respondOK(resp, modifiedReq, nextCtx)
+		if proxy.OKSendsServerTiming {
+			millis := fmt.Sprintf("dialupstream;dur=%d", time.Since(start)/time.Millisecond)
+			resp.Header.Add(serverTimingHeader, millis)
+		}
+
```

Comment:
Yeah it looks like this is primarily measuring the upstream connect and response times right @joesis? I wonder if we could tap into the data in the tracing context if/when we add that, or somehow sync time measurements between client and server...

Reply:
Oh I see sorry I misunderstood -- this would allow the client to subtract the upstream time right? Cool I like it...